src/Components/Trainers/index.js

```diff
@@ -1,11 +1,315 @@
+import { useEffect, useState } from 'react';
 import styles from './trainers.module.css';
 
-function Trainers() {
+const Trainers = () => {
+  const [trainers, setTrainers] = useState([]);
+  const [toggleAdd, setToggleAdd] = useState(false);
+  const [toggleEdit, setToggleEdit] = useState(false);
+  const [currentId, setCurrentId] = useState('');
+  const [errorMesagge, setErrorMessage] = useState('');
+  const [trainer, setTrainer] = useState({
+    firstName: '',
+    lastName: '',
+    dni: '',
+    phone: '',
+    email: '',
+    city: '',
+    password: '',
+    salary: ''
+  });
+
+  const getTrainers = async () => {
+    const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers`);
+    const data = await response.json();
+    setTrainers(data.data);
+  };
+
+  const deleteTrainer = async (_id) => {
+    const option = confirm('Are you sure??');
+    if (option) {
+      try {
+        const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/${_id}`, {
+          method: 'DELETE'
+        });
+        const data = await response.json();
+        if (!data.error) {
+          setTrainers([...trainers.filter((trainer) => trainer._id !== _id)]);
+        } else {
+          throw data.message;
+        }
+      } catch (error) {
+        alert(error);
+      }
+    }
+  };
+
+  useEffect(() => {
+    getTrainers();
+  }, []);
+
+  const addTrainer = ({ _id, firstName, lastName, dni, phone, email, city, password, salary }) => {
+    const newTrainer = {
+      _id,
+      firstName,
+      lastName,
+      dni,
+      phone,
+      email,
+      city,
+      password,
+      salary
+    };
+    setTrainers([...trainers, newTrainer]);
+  };
+
+  const sendTrainer = async (item) => {
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/trainers/`, {
+        method: 'POST',
+        body: JSON.stringify(item),
+        headers: {
+          'Content-Type': 'application/json'
+        }
+      });
+      const data = await response.json();
```

Comment:
you can make data as an object to use message, data and error without calling data.data/data.message/data.error